index.html

```diff
@@ -828,14 +828,21 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
 handler for the <dfn>resourcetimingbufferfull</dfn> event described
 below.</p>
 
+<p>To check if <dfn>can add resource timing entry</dfn>, run the following
+steps:</p>
+<ol>
+<li>If <a>resource timing buffer size limit</a> is smaller than <a>resource
+timing buffer current size</a>, return true.</li>
+<li>Return false.</li>
+</ol>
+
 <p>To <dfn>add a PerformanceResourceTiming entry</dfn> into the <a data-cite=
 'PERFORMANCE-TIMELINE-2/#dfn-performance-entry-buffer'>performance
 entry buffer</a>, run the following steps:</p>
 <ol>
 <li>Let <i>new entry</i> be the input <a>PerformanceEntry</a> to be added.
-<li>If <a>resource timing buffer current size</a> is less than
-<a>resource timing buffer size limit</a>, run the following
-substeps:
+<li>If <a>can add resource timing entry</a> returns true and <a>resource
```

Comment:
This does not solve the problem of the example I mentioned earlier:

- Buffer becomes full and we queue the buffer full task.
- We receive more entries, so they are added to the secondary buffer.
- We clear the primary buffer, thus the buffer full flag becomes false. Secondary buffer still has entries.
- We receive a new entry.

Per the current algorithm, this new entry would be added to the primary buffer before the ones in the secondary have had a chance to go into the primary buffer. So, we should check secondary buffer. If it has entries, then add the new entry to the secondary buffer.

Reply:
OK, changed the semantics of the "buffer full flag" to be "buffer full event pending flag", which resolves this issue. After discussing with @npm1, they managed to find a scenario where this is actually observable to make it testable. Should be all good now.